Zend/zend_ini.c

```diff
@@ -536,6 +538,75 @@ ZEND_API bool zend_ini_parse_bool(zend_string *str)
 	}
 }
 
+ZEND_API zend_long zend_ini_parse_quantity(zend_string *value, zend_string **errstr) /* {{{ */
+{
+	char *digits_end = NULL;
+	char *str = ZSTR_VAL(value);
+	size_t str_len = ZSTR_LEN(value);
+
+	/* Ignore trailing whitespace */
+	while (str_len && ZEND_IS_WHITESPACE(str[str_len-1])) --str_len;
+	if (!str_len) {
+		*errstr = NULL;
+		return 0;
+	}
+
+	/* Perform following multiplications on unsigned to avoid overflow UB.
+	 * For now overflow is silently ignored -- not clear what else can be
+	 * done here, especially as the final result of this function may be
+	 * used in an unsigned context (e.g. "memory_limit=3G", which overflows
+	 * zend_long on 32-bit, but not size_t). */
```

Comment:
It would be great if we could warn on overflow and on truncation to 32-bit. Detecting overflow is pretty simple. https://stackoverflow.com/a/1815371/1320374

Reply:
Definitely. I wanted to do this just after this PR, but I will do it now